sensor/common/registry/registry_store_test.go

```diff
@@ -35,194 +42,316 @@ func alwaysInsecureCheckTLS(_ context.Context, _ string) (bool, error) {
+		assert.Len(t, regStore.storeByName, 1, "non-global store should have also had an upsert")
+	})
 }
 
+// TestRegistryStore_GlobalStoreLazyNoFailUpsertCheckTLS mirrors the objective of
+// the LazyNoFailUpsertCheckTLS test but instead upserts global registries.
 func TestRegistryStore_GlobalStoreLazyNoFailUpsertCheckTLS(t *testing.T) {
-	testutils.MustUpdateFeature(t, features.SensorLazyTLSChecks, true)
+	t.Setenv(env.DelegatedScanningDisabled.EnvVar(), "false")
+
+	// commonTests executes tests that should have the same results regardless of
+	// how secrets are stored.
+	commonTests := func(t *testing.T) {
+		regStore := NewRegistryStore(alwaysFailCheckTLS)
+		dce := config.DockerConfigEntry{Username: "username", Password: "password"}
+		dc := config.DockerConfig{fakeImgName.GetRegistry(): dce}
+
+		// Upsert should NOT fail on lazy TLS check
+		regStore.UpsertSecret(openshiftConfigNamespace, openshiftConfigPullSecret, dc, "")
+		require.False(t, regStore.globalRegistries.IsEmpty())
+		allRegs := regStore.globalRegistries.GetAll()
+		require.Len(t, allRegs, 1)
+	}
+
+	t.Run("SecretsByHost", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, false)
 
-	ctx := context.Background()
+		commonTests(t)
+	})
+
+	t.Run("SecretsByName", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, true)
+
+		commonTests(t)
+	})
+}
+
+// TestRegistryStore_CentralIntegrations test the upsert, delete, and retrieval
+// of registries built from image integrations synced to Sensor from Central.
+func TestRegistryStore_CentralIntegrations(t *testing.T) {
 	regStore := NewRegistryStore(alwaysFailCheckTLS)
-	dce := config.DockerConfigEntry{Username: "username", Password: "password"}
 
-	// upsert that fails TLS check should error out and NOT perform an upsert
-	require.NoError(t, regStore.UpsertGlobalRegistry(ctx, fakeImgName.GetRegistry(), dce))
-	require.False(t, regStore.globalRegistries.IsEmpty())
-	allRegs := regStore.globalRegistries.GetAll()
-	require.Len(t, allRegs, 1)
+	iis := []*storage.ImageIntegration{
+		{Id: "bad", Name: "bad", Type: "bad"},
+		{Id: "a", Name: "a", Type: types.DockerType, IntegrationConfig: &storage.ImageIntegration_Docker{}},
+		{Id: "b", Name: "b", Type: types.DockerType, IntegrationConfig: &storage.ImageIntegration_Docker{}},
+		{Id: "c", Name: "c", Type: types.DockerType, IntegrationConfig: &storage.ImageIntegration_Docker{
+			Docker: &storage.DockerConfig{Endpoint: "example.com"}},
+		},
+	}
+
+	regStore.UpsertCentralRegistryIntegrations(iis)
+	assert.Len(t, regStore.centralRegistryIntegrations.GetAll(), 3)
+
+	regStore.DeleteCentralRegistryIntegrations([]string{"a", "b"})
+	assert.Len(t, regStore.centralRegistryIntegrations.GetAll(), 1)
+
+	imgName, _, err := utils.GenerateImageNameFromString("example.com/repo/path:tag")
+	require.NoError(t, err)
+	regs := regStore.GetCentralRegistries(imgName)
```

Comment:
nit: would be nice to test this before the deletions since there is only one choice now. Prior to deletions there were 3